linera-base/src/crypto/mod.rs

```diff
@@ -27,11 +31,139 @@ pub type ValidatorKeypair = secp256k1::Secp256k1KeyPair;
 /// The public key of a chain owner.
 /// The corresponding private key is allowed to propose blocks
 /// on the chain and transfer account's tokens.
-pub type AccountPublicKey = ed25519::Ed25519PublicKey;
+#[derive(
+    Serialize,
+    Deserialize,
+    Debug,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Copy,
+    Clone,
+    Hash,
+    WitType,
+    WitLoad,
+    WitStore,
+)]
+pub struct AccountPublicKey(ed25519::Ed25519PublicKey);
```

Comment:
Is this meant to become an enum when we support several algorithms?

Reply:
Yes, exactly.